### ISSUE TITLE
-special-function-bound-interpolation causes over subsumption

### ISSUE THREAD

**Issue by domainexpert** [tracer-x/klee#260]
This is demonstrated  by `basic/pointer_wp3.c` of `klee-examples` repository, which loses error report when -special-function-bound-interpolation is specified. Thanks to @rasoolmaghareh for reporting the problems that led to this issue.

**Comment by domainexpert** [comment]
Resolved via #261 .

**Comment by rasoolmaghareh** [comment]
@domainexpert, I have noticed that Regexp10.c with bound 29 fails with a couterexample of length 35. However, it succeeds with bound 32 (it should fail with counterexample of length 35).
```
/*
 * Simple regular expression matching.
 *
 * From:
 *   The Practice of Programming
 *   Brian W. Kernighan, Rob Pike
 *
 * This code with KLEE harness is obtained from the following
 * page of the KLEE tutorial on December 2015:
 *   http://klee.github.io/resources/Regexp.c.html
 */
#ifdef LLBMC
#include <llbmc.h>
#else
#include <klee/klee.h>
#endif

char _bound[29];
char *wcet;
void tracerx_check(char *p) { printf("Timing of Path:%d\n",(int)p-(int)(&(_bound))); *p;}

static int matchhere(char*,char*);

static int matchstar(int c, char *re, char *text) {
  do {
    if (matchhere(re, text)){
      wcet++;	
      return 1;
    }
  } while (*text != '\0' && (*text++ == c || c== '.'));
  return 0;
}

static int matchhere(char *re, char *text) {
  if (re[0] == '\0'){
     wcet++;
     return 0;
  }
  if (re[1] == '*'){
    return matchstar(re[0], re+2, text);
    wcet++;
  }
  if (re[0] == '$' && re[1]=='\0'){
    wcet++;
    return *text == '\0';
  }
  if (*text!='\0' && (re[0]=='.' || re[0]==*text)){
    wcet++;
    return matchhere(re+1, text+1);
  }
  return 0;
}

int match(char *re, char *text) {
  if (re[0] == '^'){
    wcet++;
    return matchhere(re+1, text);
  }
  do {
    if (matchhere(re, text)){
      wcet++;
      return 1;
    }
  } while (*text++ != '\0');
  return 0;
}

/*
 * Harness for testing with KLEE.
 */

// The size of the buffer to test with.
#define SIZE 10

int main() {
  wcet=_bound;
  // The input regular expression.
  char re[SIZE];
  wcet++;

// Make the input symbolic.
#ifdef LLBMC
  for (int i = 0; i < SIZE; ++i) {
    re[i] = __llbmc_nondef_char();
  }
#else
  klee_make_symbolic(re, sizeof re, "re");
  klee_assume(re[SIZE-1] == '\0');
#endif

  // Try to match against a constant string "hello".
  match(re, "hello");
  tracerx_check(wcet);
  return 0;
}
```

**Comment by domainexpert** [comment]
@rasoolmaghareh This is more like a problem that KLEE found a error with size 32 of `_bound` and Tracer-X did not.

**Comment by domainexpert** [comment]
@rasoolmaghareh I confirm that this is happening at bound size 26 the lowest.

**Comment by domainexpert** [comment]
@rasoolmaghareh Bound size 10 is the lowest I found discrepancy in the result of KLEE (violating at 10) and Tracer-X (violating at 11), both were run using DFS.

**Comment by rasoolmaghareh** [comment]
My checks show that the tighter the bound gets, the higher will be the chance of Tracer-X not missing the bound which is expected based on our understanding of bound propagation.

**Comment by domainexpert** [comment]
@rasoolmaghareh This is the example with smallest bounds where Tracer-X shows difference in violating value to KLEE (8 vs. 7).
```
/*
 * Simple regular expression matching.
 *
 * From:
 *   The Practice of Programming
 *   Brian W. Kernighan, Rob Pike
 *
 * This code with KLEE harness is obtained from the following
 * page of the KLEE tutorial on December 2015:
 *   http://klee.github.io/resources/Regexp.c.html
 */
#ifdef LLBMC
#include <llbmc.h>
#else
#include <klee/klee.h>
#endif

char _bound[7];
char *wcet;
void tracerx_check(char *p) { printf("Timing of Path:%d\n",((uint64_t)p)-((uint64_t)(&(_bound)))); *p;}

static int matchhere(char*,char*);

static int matchstar(int c, char *re, char *text) {
  do {
    if (matchhere(re, text)){
      wcet++;	
      return 1;
    }
  } while (*text != '\0' && (*text++ == c || c== '.'));
  return 0;
}

static int matchhere(char *re, char *text) {
  if (re[0] == '\0'){
     wcet++;
     return 0;
  }
  if (re[1] == '*'){
    return matchstar(re[0], re+2, text);
    wcet++;
  }
  if (re[0] == '$' && re[1]=='\0'){
    wcet++;
    return *text == '\0';
  }
  if (*text!='\0' && (re[0]=='.' || re[0]==*text)){
    wcet++;
    return matchhere(re+1, text+1);
  }
  return 0;
}

int match(char *re, char *text) {
  if (re[0] == '^'){
    wcet++;
    return matchhere(re+1, text);
  }
  do {
    if (matchhere(re, text)){
      wcet++;
      return 1;
    }
  } while (*text++ != '\0');
  return 0;
}

/*
 * Harness for testing with KLEE.
 */

// The size of the buffer to test with.
#define SIZE 4

int main() {
  wcet=_bound;
  // The input regular expression.
  char re[SIZE];
  wcet++;

// Make the input symbolic.
#ifdef LLBMC
  for (int i = 0; i < SIZE; ++i) {
    re[i] = __llbmc_nondef_char();
  }
#else
  klee_make_symbolic(re, sizeof re, "re");
  klee_assume(re[SIZE-1] == '\0');
#endif

  // Try to match against a constant string "hello".
  match(re, "hello");
  tracerx_check(wcet);
  return 0;
}
```

**Comment by domainexpert** [comment]
Resolved via #268.
